experimental/displaynames/src/displaynames.rs

```diff
@@ -330,29 +333,80 @@ impl LanguageDisplayNames {
 /// )
 /// .expect("Data should load successfully");
 ///
-/// assert_eq!(display_name.of(&locale!("de-CH")), "Swiss High German");
-/// assert_eq!(display_name.of(&locale!("de")), "German");
-/// assert_eq!(display_name.of(&locale!("de-MX")), "German (Mexico)");
-/// assert_eq!(display_name.of(&locale!("xx-YY")), "xx (YY)");
-/// assert_eq!(display_name.of(&locale!("xx")), "xx");
+/// assert_eq!(display_name.of(&locale!("en-GB")), "British English");
+/// assert_eq!(display_name.of(&locale!("en")), "English");
+/// assert_eq!(display_name.of(&locale!("en-MX")), "English (Mexico)");
 /// ```
 pub struct LocaleDisplayNamesFormatter {
     options: DisplayNamesOptions,
     // patterns: DataPayload<LocaleDisplayNamesPatternsV1Marker>,
     locale_data: DataPayload<LocaleDisplayNamesV1Marker>,
 
     language_data: DataPayload<LanguageDisplayNamesV1Marker>,
-    #[allow(dead_code)] // TODO use this
     script_data: DataPayload<ScriptDisplayNamesV1Marker>,
     region_data: DataPayload<RegionDisplayNamesV1Marker>,
-    #[allow(dead_code)] // TODO add support for variants
     variant_data: DataPayload<VariantDisplayNamesV1Marker>,
     // key_data: DataPayload<KeyDisplayNamesV1Marker>,
     // measuerment_data: DataPayload<MeasurementSystemsDisplayNamesV1Marker>,
     // subdivisions_data: DataPayload<SubdivisionsDisplayNamesV1Marker>,
     // transforms_data: DataPayload<TransformsDisplayNamesV1Marker>,
 }
 
+// LongestMatching subtag is a longest substring of a given locale that exists as a key in the CLDR locale data.
+// This is used for implementing Locale Display Name Algorithm.
+#[derive(PartialEq, Clone, Copy)]
+enum LongestMatchingSubtag {
+    // Longest matching subtag of type ${lang}-${region}.
+    // Example: "de-ET", "en-GB"
+    LangRegion,
+    // Longest matching subtag of type ${lang}-${script}.
+    // Example: "hi-Latn", "zh-Hans"
+    LangScript,
+    // Longest matching subtag of type ${lang}
+    // Example: "en", "hi"
+    Lang,
```

Comment:
Please note why LangScriptRegion isn't in this list

Reply:
This enum is not needed anymore based on a suggestion from Robert.